source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -790,17 +949,21 @@ DefaultCredentialsProviderChain::DefaultCredentialsProviderChain(
                 full_uri);
       add(factories.createContainerCredentialsProvider(
           api, context, fetch_metadata_using_curl, MetadataFetcher::create,
-          CONTAINER_METADATA_CLUSTER, full_uri, authorization_token));
+          CONTAINER_METADATA_CLUSTER, full_uri, receiver_state, initialization_timer,
+          authorization_token));
     } else {
       ENVOY_LOG(debug, "Using container role credentials provider with URI: {}", full_uri);
-      add(factories.createContainerCredentialsProvider(api, context, fetch_metadata_using_curl,
-                                                       MetadataFetcher::create,
-                                                       CONTAINER_METADATA_CLUSTER, full_uri));
+      add(factories.createContainerCredentialsProvider(
+          api, context, fetch_metadata_using_curl, MetadataFetcher::create,
+          CONTAINER_METADATA_CLUSTER, full_uri, receiver_state, initialization_timer));
     }
-  } else if (metadata_disabled != TRUE) {
+  }
+
+  if (metadata_disabled != TRUE) {
```

Comment:
The reason to have `else if` I believe is to use ECS task role if available on ECS and not to fall back to IMDS in any scenario.

Any reason to think otherwise where, we have fall back on IMDS if we don't get credentials from Task role on ECS containers? I have to confirm the behavior from AWS SDK what we do when we run inside ECS containers.